### PR TITLE
Add Walker and Display for Promise objects under debugger locals

### PR DESCRIPTION
**Pull request by agarwal-sandeep**
Promise objects will now show [Promise] fake node with [status] and
[value] as childrens.

[status] is one of undefined/pending/resolved/rejected
[value] is either undefined or the value display.

### CODE REVIEW

**Comment by agarwal-sandeep** [comment]
Should fix #671 when this merges with JsRTDebugging changes #926

**Comment by agarwal-sandeep** [comment]
@boingoing @akroshg please take a look

File: lib/Runtime/Debug/DiagObjectModel.h [hunk 1]
```diff
@@ -867,6 +867,25 @@ namespace Js
...
+        virtual WeakArenaReference<IDiagObjectModelWalkerBase>* CreateWalker() override;
+    };
+
+    class RecyclablePromiseObjectWalker : public RecyclableObjectWalker
+    {
+    public:
+        RecyclablePromiseObjectWalker(ScriptContext* pContext, Var instance);
+
+        virtual BOOL Get(int i, ResolvedObject* pResolvedObject) override;
+        virtual ulong GetChildrenCount() override { return 2; }
```
Comment (akroshg) [inline lib/Runtime/Debug/DiagObjectModel.h]: nit: could you comment here on readability on why 2?

File: lib/Runtime/Debug/DiagObjectModel.cpp [hunk 2]
```diff
@@ -3657,6 +3667,101 @@ namespace Js
...
+            Assert(false);
+            return FALSE;
+        }
+
+        pResolvedObject->propId = Constants::NoProperty;
+        pResolvedObject->scriptContext = scriptContext;
+        pResolvedObject->typeId = JavascriptOperators::GetTypeId(pResolvedObject->obj);
+        pResolvedObject->objectDisplay = pResolvedObject->CreateDisplay();
+        pResolvedObject->objectDisplay->SetDefaultTypeAttribute(DBGPROP_ATTRIB_VALUE_READONLY | DBGPROP_ATTRIB_VALUE_IS_FAKE);
+        pResolvedObject->address = Anew(GetArenaFromContext(pResolvedObject->scriptContext),
```
Comment (akroshg) [inline lib/Runtime/Debug/DiagObjectModel.cpp]: above fields ([status] and [value]) are not editable - why do you need to populate the address?

File: lib/Runtime/Library/JavascriptPromise.h [hunk 3]
```diff
@@ -372,11 +371,16 @@ namespace Js
             PromiseStatusCode_HasRejection
         };
 
+    protected:
         PromiseStatus status;
         Var result;
         JavascriptPromiseReactionList* resolveReactions;
         JavascriptPromiseReactionList* rejectReactions;
 
+    public:
```
Comment (akroshg) [inline lib/Runtime/Library/JavascriptPromise.h]: nit: move it up to be with other methods?

**Comment by akroshg** [comment]
overall looks good to me.

**Pull request merged**